src/network/index.ts

```diff
@@ -0,0 +1,145 @@
+  }
+
+  private constructor(apiUrl: ApiUrl, appDetails: AppDetails, opts: { bridgeUser: string; userId: string }) {
+    this.client = HttpClient.create(apiUrl);
+    this.appDetails = appDetails;
+    this.auth = {
+      username: opts.bridgeUser,
+      password: opts.userId,
+    };
+  }
+
+  public startUpload(idBucket: string, payload: StartUploadPayload): Promise<StartUploadResponse> {
+    for (const { index, size } of payload.uploads) {
+      if (!isHexString(index) || index.length !== 64) {
+        throw new Error('Invalid index');
+      }
+      if (size < 0) {
+        throw new Error('Invalid size');
+      }
+    }
+    return Network.startUpload(idBucket, payload, {
+      client: this.client,
+      appDetails: this.appDetails,
+      auth: this.auth,
+    });
+  }
+
+  public finishUpload(idBucket: string, payload: FinishUploadPayload): Promise<FinishUploadResponse> {
+    const { index, shards } = payload;
+    if (!isHexString(index) || index.length !== 64) {
+      throw new Error('Invalid index');
+    }
+
+    for (const shard of shards) {
+      if (!uuidValidate(shard.uuid)) {
+        throw new Error('Invalid UUID');
+      }
+    }
+
+    return Network.finishUpload(idBucket, payload, {
+      client: this.client,
+      appDetails: this.appDetails,
+      auth: this.auth,
+    });
+  }
+
+  public getDownloadLinks(idBucket: string, file: string): Promise<GetDownloadLinksResponse> {
+    return Network.getDownloadLinks(idBucket, file, {
+      client: this.client,
+      appDetails: this.appDetails,
+      auth: this.auth,
+    });
+  }
+
+  public async deleteFile(idBucket: string, file: string): Promise<void> {
+    await Network.deleteFile(idBucket, file, {
+      client: this.client,
+      appDetails: this.appDetails,
+      auth: this.auth,
+    });
+  }
+
+  /**
+   * Creates entries for every upload in the request, returns the urls to upload
+   * @param idBucket
+   * @param uploads
+   */
+  private static startUpload(
+    idBucket: string,
+    payload: StartUploadPayload,
+    { client, appDetails, auth }: NetworkRequestConfig,
+  ) {
+    const headers = Network.headersWithBasicAuth(appDetails, auth);
+    return client.post<StartUploadResponse>(`/v2/buckets/${idBucket}/files/start`, payload, headers);
+  }
+
+  /**
+   * Finishes the upload of a file
+   * @param idBucket
+   * @param index
+   * @param shards
+   */
+  private static finishUpload(
+    idBucket: string,
+    payload: FinishUploadPayload,
+    { client, appDetails, auth }: NetworkRequestConfig,
+  ) {
+    const headers = Network.headersWithBasicAuth(appDetails, auth);
+    return client.post<FinishUploadResponse>(`/v2/buckets/${idBucket}/files/finish`, payload, headers);
+  }
+
+  /**
+   * Gets the download links for a file
+   * @param idBucket
+   * @param file
+   */
+  private static getDownloadLinks(idBucket: string, file: string, { client, appDetails, auth }: NetworkRequestConfig) {
```

Comment:
Same here (related to the variable naming)

Reply:
fixed